src/main/java/xyz/nucleoid/bedwars/game/active/shop/BwItemShop.java

```diff
@@ -12,24 +12,209 @@
 import xyz.nucleoid.plasmid.shop.ShopUi;
 import xyz.nucleoid.plasmid.util.ColoredBlocks;
 import xyz.nucleoid.plasmid.util.ItemStackBuilder;
+
+import org.apache.commons.compress.archivers.cpio.CpioArchiveEntry;
+import org.apache.commons.compress.archivers.dump.DumpArchiveConstants.COMPRESSION_TYPE;
+
+import net.minecraft.block.Block;
 import net.minecraft.block.Blocks;
+import net.minecraft.datafixer.fix.ItemSpawnEggFix;
 import net.minecraft.enchantment.Enchantments;
+import net.minecraft.entity.effect.StatusEffectInstance;
+import net.minecraft.entity.effect.StatusEffects;
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.Items;
+import net.minecraft.potion.Potion;
+import net.minecraft.potion.PotionUtil;
+import net.minecraft.potion.Potions;
+import net.minecraft.resource.ResourceManager.Empty;
+import net.minecraft.server.command.EffectCommand;
 import net.minecraft.server.network.ServerPlayerEntity;
 import net.minecraft.text.LiteralText;
 import net.minecraft.text.Text;
 import net.minecraft.util.DyeColor;
 
 public final class BwItemShop {
-    //TODO: reorganize shop layout
-    public static ShopUi create(ServerPlayerEntity player, BwActive game) {
+    public static ShopUi create(ServerPlayerEntity player, BwActive game, int pageIndex) {
```

Comment:
It might be worth having a different `create` method for each page number instead of the if/else chain

Reply:
Is this function taking an index still needed now that the specific methods could be called directly?

Reply:
It's no longer needed, changing now